docs/configuration/secure.md

```diff
@@ -0,0 +1,571 @@
+    ...
+    ServerName yourservername.org
+    ...
+    SSLEngine on
+    SSLCertificateFile       /path/to/signed_cert_plus_intermediates
+    SSLCertificateKeyFile   /path/to/private/key
+```
+
+Add the following line OUTSIDE  the `` <VirtualHost></VirtualHost> `` block:
+```
+SSLProtocol             all -SSLv3 -TLSv1 -TLSv1.1
+SSLCipherSuite          ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256
+SSLHonorCipherOrder     on
+SSLCompression          off
+SSLSessionTickets       off
+```
+
+
+#### NGINX
+*(Version: 1.14.0,  OpenSSL 1.1.1c)*
+
+Add the following line INSIDE the server block:
+
+```
+server {
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+    ...
+    server_name yourservername.org;
+    ...
+    # certs sent to the client in SERVER HELLO are concatenated in ssl_certificate
+    ssl_certificate /path/to/signed_cert_plus_intermediates;
+    ssl_certificate_key /path/to/private_key;
+    ssl_session_timeout 5m;
+    ssl_session_cache shared:SSL:16m;
+    ssl_session_tickets off;
+
+    ssl_protocols TLSv1.2;
+    ssl_ciphers 'ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256';
+    ssl_prefer_server_ciphers on;
+
+    ssl_dhparam /etc/ssl/private/dhparam.pem;
+
+
+```
+
+``ssl_dhparam /etc/ssl/private/dhparam.pem;``:
+
+This specifies a file with DH parameters for EDH (Ephemeral Diffie-Hellman) ciphers.  
+By default, NGINX will use the default DHE paramaters provided by openssl. This uses a weak key that gets lower scores.  
+Run ``openssl dhparam -out /etc/ssl/private/dhparam.pem 4096`` in terminal to generate it.
+
+We RECOMMEND to use the [Mozilla SSL Configuration Generator](https://ssl-config.mozilla.org/) to generate a suitable configuration and the [Qualys SSL Labs Tests](https://www.ssllabs.com/ssltest/) or the [High-Tech Bridge SSL Server Test](https://www.htbridge.com/ssl/) to check your settings. It is recommended, to reach a "A" rating as minimum.
+
+
+### Serve security related Headers
+
+To improve the security of your ILIAS users you SHOULD set the following Headers:
+
+
+```
+    X-Content-Type-Options: nosniff;
+    X-XSS-Protection: 1; mode=block;
+    X-Frame-Options: SAMEORIGIN;
+    Referrer-Policy: strict-origin;
+    Feature-Policy sync-xhr 'self';  
+    add_header Content-Security-Policy "default-src 'self'; connect-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval' blob:; style-src 'self' 'unsafe-inline' data:; img-src 'self' 'unsafe-inline' data:; font-src 'self' 'unsafe-inline' data:; media-src 'self' 'unsafe-inline' data:";  
```

Comment:
I wouldn't make this CSP recommendation for the same reasons as corro. A link to https://observatory.mozilla.org should be enough. If we really want to use CSP we should avoid the need for "unsafe-inline" completely, otherwise it doesn't make much sense.

Reply:
With the requirement for the secure document it is necessary to give admins a detailed configuration suggestion.

As a result of discussion, we provide a more or sometimes less "best practice" configuration.